File: Java/Easy/Accepted/reformat-date_8.java
class Solution {
    public String reformatDate(String date) {
        
        String[] split = date.split(" ");
        String day = split[0];
        String month = split[1];
        String year = split[2];
        String[] days = new String[]{"1st", "2nd", "3rd", "4th", "5th", "6th", "7th", "8th", "9th", "10th", "11th", "12th", "13th", "14th", "15th", "16th", "17th", "18th", "19th", "20th", "21st", "22nd", "23rd", "24th", "25th", "26th", "27th", "28th", "29th", "30th", "31st"};
        String[] months = new String[]{"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
        String dayNum = day.substring(0, day.length() - 2);
        String monthNum = "";
        for (int i = 0; i < months.length; i++) {
            if (month.equals(months[i])) {
                monthNum = String.format("%02d", i + 1);
            }
        }
        return year + "-" + monthNum + "-" + String.format("%02d", Integer.parseInt(dayNum));
    }
}

File: Rust/Easy/CompileError/reformat-date_8.rs
impl Solution {
    pub fn reformat_date(date: String) -> String {
        
        let mut month = String::new();
        let mut day = String::new();
        let mut year = String::new();
        let mut month_num = 0;
        let mut day_num = 0;
        let mut year_num = 0;
        let mut start = 0;
        let mut end = 0;
        let mut i = 0;
        for c in date.chars() {
            if c >= '0' && c <= '9' {
                if start == 0 {
                    start = i;
                }
                end = i;
            } else if c >= 'a' && c <= 'z' {
                if start == 0 {
                    start = i;
                }
                end = i;
            } else if c >= 'A' && c <= 'Z' {
                if start == 0 {
                    start = i;
                }
                end = i;
            }
            i += 1;
        }
        day = date[start..=end].to_string();
        //println!("{}", day);
        start = 0;
        end = 0;
        i = 0;
        for c in date.chars() {
            if c >= '0' && c <= '9' {
                if start == 0 {
                    start = i;
                }
                end = i;
            } else if c >= 'a' && c <= 'z' {
                if start == 0 {
                    start = i;
                }
                end = i;
            } else if c >= 'A' && c <= 'Z' {
                if start == 0 {
                    start = i;
                }
                end = i;
            }
            i += 1;
        }
        month = date[start..=end].to_string();
        //println!("{}", month);
        start = 0;
        end = 0;
        i = 0;
        for c in date.chars() {
            if c >= '0' && c <= '9' {
                if start == 0 {
                    start = i;
                }
                end = i;
            } else if c >= 'a' && c <= 'z' {
                if start == 0 {
                    start = i;
                }
                end = i;
            } else if c >= 'A' && c <= 'Z' {
                if start == 0 {
                    start = i;
                }
    }
}